Packs/PassiveTotal/Integrations/PassiveTotal_v2/PassiveTotal_v2.py

```diff
@@ -0,0 +1,871 @@
+
+        if status_code != 200:
+            error_message = ''
+            if resp.json().get('message', ''):
+                error_message = 'Reason: {}'.format(resp.json().get('message', ''))
+            status_code_message_map = {
+                400: '{} {}'.format(MESSAGES['BAD_REQUEST_ERROR'], error_message),
+                401: MESSAGES['AUTHENTICATION_ERROR'],
+                404: MESSAGES['PAGE_NOT_FOUND_ERROR'],
+                407: MESSAGES['PROXY_ERROR']
+            }
+            if status_code in status_code_message_map:
+                LOG('Response code: {}. Reason: {}'.format(status_code, status_code_message_map[status_code]))
+                raise ValueError(status_code_message_map[status_code])
+            elif status_code >= 500:
+                LOG('Response code: {}. Reason: {}'.format(status_code, MESSAGES['INTERNAL_SERVER_ERROR']))
+                raise ValueError(MESSAGES['INTERNAL_SERVER_ERROR'])
+            else:
+                resp.raise_for_status()
+
+        return resp.json()
+
+
+''' HELPER FUNCTIONS '''
+
+
+def get_request_timeout() -> Optional[Any]:
+    """
+    Validate and return the request timeout parameter.
+    The parameter must be a positive integer.
+    Default value is set to 20 seconds for API request timeout.
+
+    :params req_timeout: Request timeout value.
+    :return: boolean
+    """
+    try:
+        request_timeout = demisto.params().get('request_timeout', DEFAULT_REQUEST_TIMEOUT)
+        request_timeout = DEFAULT_REQUEST_TIMEOUT if not request_timeout else request_timeout
+        request_timeout = int(request_timeout)
+    except ValueError:
+        raise ValueError(MESSAGES['REQUEST_TIMEOUT_VALIDATION'])
+
+    if request_timeout <= 0:
+        raise ValueError(MESSAGES['REQUEST_TIMEOUT_VALIDATION'])
+    elif request_timeout > REQUEST_TIMEOUT_MAX_VALUE:
+        raise ValueError(MESSAGES['REQUEST_TIMEOUT_EXCEED_ERROR'])
+
+    return request_timeout
+
+
+def get_host_attribute_context_data(records: List[Dict[str, Any]]) -> Tuple[list, list]:
+    """
+    Prepares context data for get components and get trackers command
+
+    :param records: API response
+    :return: standard entry context and custom entry context
+    """
+    standard_ec: List[Union[Common.Domain, Common.IP]] = []
+
+    for record in records:
+        if record.get('hostname'):
+            hostname = record.get('hostname')
+            dbot_score = Common.DBotScore(indicator=hostname, indicator_type=DBotScoreType.DOMAIN,
+                                          integration_name=INTEGRATION_NAME, score=Common.DBotScore.NONE)
+            standard_ec.append(Common.Domain(domain=hostname, dbot_score=dbot_score))
+        elif record.get('address'):
+            address = record.get('address')
+            dbot_score = Common.DBotScore(indicator=address, indicator_type=DBotScoreType.IP,
+                                          integration_name=INTEGRATION_NAME, score=Common.DBotScore.NONE)
```

Comment:
Because it is mandatory to create it. Ref: https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L2369